pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -209,20 +209,26 @@ func (c *controller) checkReadyForEvents(curr interface{}) error {
 }
 
 func (c *controller) tryLedgerPut(obj interface{}, schema collection.Schema) {
-	iobj := obj.(crd.IstioObject)
+	iobj, err := castToIstioObject(obj)
+	if err != nil {
+		scope.Errorf("Failed to cast to crd.IstioObject in ledger put: %v", err)
+	}
```

Comment:
Shouldn't there be a return or something here?  the object failed to cast, so the var will be nil, which is probably what's causing the panic in the unit test.

Reply:
Oops. Yes, of course. Will push a fix. Thanks!